### ISSUE TITLE
Upcoming breaking changes in Stylelint 17.0.0

### ISSUE THREAD

**Issue by jeddy3** [stylelint-scss/stylelint-scss#1215]
@kristerkari We wanted to give you a heads up about [some breaking changes coming in Stylelint 17.0.0](https://github.com/stylelint/stylelint/blob/8b509b7b5db529cf01595f0e9e089d981a68ee54/docs/migration-guide/to-17.md).

We've tried to keep them to a minimum, but some will affect `stylelint-scss`:

- removed the CommonJS Node.js API and Stylelint 17.0.0 will be an ES Module (ESM)
- change to standard spec-based nesting in a number of rules

This package will need to be an ESM to support the former (which was started in https://github.com/stylelint-scss/stylelint-scss/pull/931).

For the latter, we've [removed the `resolveNestedSelectors` option from `selector-class-pattern`](https://github.com/stylelint/stylelint/pull/8931) because it's incompatible with standard nesting, which doesn't concatenate strings. 

For example, this is valid in SCSS but invalid in CSS:

```css
.foo {
  &__bar {}
}
```

In the migration guide, we're encouraging the SCSS community to create their own plugin version of the rule. It would use nonstandard nesting under the hood to reinstate the option. It seems like this option is quite widely used SCSS authors. It can be done by repurposing [the removed](https://github.com/stylelint/stylelint/pull/8931) `flattenNestedSelectorsForRule.mjs‎` util that uses the `postcss-resolve-nested-selector` package to resolve selectors.

We've also changed the following rules to use standard nesting, which often desugars to `:is()` and therefore calculates specificity differently than if SCSS desugared them:

- `selector-max-specificity`
- `no-descending-specificity` 
- `no-duplicate-selectors`
- `selector-no-qualify-type`

For example:

```css
#foo
.bar {
  a {}
}
```

Is desugared in standard CSS to:

```css
:is(#foo, .bar) a {}
```

And the following in SCSS:

```css
#foo a {}
.bar a {}
```

We're recommending (in the migration guide and rule READMEs) that SCSS users turn off these rules in their configs, and you can do the same in the shared SCSS config, or create SCSS versions of these rules. However, the behaviour is only subtly different and not as big a change as removing the `resolveNestedSelectors` option from `selector-class-pattern`. If there's time to create only one plugin, we recommend it be a SCSS version of `selector-class-pattern`.

Feel free to drop any questions or queries here, and we'll do our best to answer them.

Thank you for supporting the SCSS community with the plugin!

**Comment by kristerkari** [comment]
> Given the number of moving parts, I'm happy to release the SCSS packages as part of our `17.0.0` release process (in a week's time or so). I believe I still have NPM access to the SCSS packages.
> 
> It's not usually something we'd do, but the change to spec-based nesting is only disruptive for SCSS users. We'll then be able to mention the packages in our migration guide, giving SCSS users a more straightforward migration path.
> 
> Would you be OK with that?

That's totally fine! Let me know if you are missing any npm package permission and I can help out.

**Comment by kristerkari** [comment]
> When you have a moment, can you also give the config PRs a once-over, please?:

Done!
